ethdb/privateapi/ethbackend.go

```diff
@@ -463,31 +546,34 @@ func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *r
 		}, nil
 	}
 
-	if s.stageLoopIsBusy() {
-		log.Debug("[ForkChoiceUpdated] stage loop is busy")
-		return &remote.EngineForkChoiceUpdatedReply{
-			PayloadStatus: &remote.EnginePayloadStatus{Status: remote.EngineStatus_SYNCING},
-		}, nil
+	status, err := s.getPayloadStatusFromHashIfPossible(forkChoice.HeadBlockHash, 0, common.Hash{}, false)
+	if err != nil {
+		return nil, err
 	}
-
-	log.Debug("[ForkChoiceUpdated] acquiring lock")
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	log.Debug("[ForkChoiceUpdated] lock acquired")
-
-	log.Debug("[ForkChoiceUpdated] sending forkChoiceMessage", "head", forkChoice.HeadBlockHash)
-	s.requestList.AddForkChoiceRequest(&forkChoice)
+	if status == nil {
+		if s.stageLoopIsBusy() {
```

Comment:
Let's move the call of `stageLoopIsBusy` before `s.lock.Lock()` since `stageLoopIsBusy` might be slow.